src/main/kotlin/com/terraformation/backend/daily/NotificationScanner.kt

```diff
@@ -0,0 +1,79 @@
+package com.terraformation.backend.daily
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.customer.db.FacilityStore
+import com.terraformation.backend.customer.event.FacilityTimeZoneChangedEvent
+import com.terraformation.backend.customer.model.FacilityModel
+import com.terraformation.backend.customer.model.SystemUser
+import com.terraformation.backend.log.perClassLogger
+import com.terraformation.backend.time.ClockAdvancedEvent
+import java.time.InstantSource
+import java.time.LocalDate
+import javax.inject.Inject
+import javax.inject.Named
+import org.jobrunr.scheduling.JobScheduler
+import org.jobrunr.scheduling.cron.Cron
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.event.EventListener
+import org.springframework.core.annotation.Order
+
+@ConditionalOnProperty(TerrawareServerConfig.DAILY_TASKS_ENABLED_PROPERTY, matchIfMissing = true)
+@Named
+class NotificationScanner(
+    private val clock: InstantSource,
+    private val config: TerrawareServerConfig,
+    private val eventPublisher: ApplicationEventPublisher,
+    private val facilityStore: FacilityStore,
+    private val notifiers: List<FacilityNotifier>,
```

Comment:
Embarrassing question.. do the named notifiers (seedbank and nursery) get bound to this list automagically?

Reply:
Totally reasonable question. Yes, Spring looks at the type argument and passes in a list of all the objects it manages that are instances of that type. Did it this way to maintain roughly the same level of decoupling we had before with the event listeners: `NotificationScanner` doesn't need to know what the notifiers actually are.